donkeycar/pipeline/types.py

```diff
@@ -71,27 +78,68 @@ class TubDataset(object):
     """
 
     def __init__(self, config: Config, tub_paths: List[str],
-                 shuffle: bool = True) -> None:
+                 seq_size: int = None) -> None:
         self.config = config
         self.tub_paths = tub_paths
-        self.shuffle = shuffle
         self.tubs: List[Tub] = [Tub(tub_path, read_only=True)
                                 for tub_path in self.tub_paths]
         self.records: List[TubRecord] = list()
         self.train_filter = getattr(config, 'TRAIN_FILTER', None)
+        self.seq_size = seq_size
+
+    def get_records(self):
+        if not self.records:
+            msg = f'Loading tubs from paths {self.tub_paths} with filter ' \
+                  f'{self.train_filter}' if self.train_filter else ''
+            logger.info(msg)
+            for tub in self.tubs:
+                for underlying in tub:
+                    record = TubRecord(self.config, tub.base_path, underlying)
+                    if not self.train_filter or self.train_filter(record):
+                        self.records.append(record)
+            if self.seq_size:
+                seq = Sequencer(self.seq_size, self.records)
+                self.records = list(seq)
+        return self.records
+
+
+class Sequencer(Iterable[List[TubRecord]]):
+    """" Builds a sequence of continuous records for RNN and similar models. """
+    def __init__(self, seq_length: int, records: List[TubRecord]):
+        """
+        :param seq_length:  length of sequence
+        :param records:     input record list
+        """
+        self.records = records
+        self.seq_length = seq_length
 
-    def train_test_split(self) -> Tuple[List[TubRecord], List[TubRecord]]:
-        msg = f'Loading tubs from paths {self.tub_paths}' + f' with filter ' \
-              f'{self.train_filter}' if self.train_filter else ''
-        print(msg)
-        self.records.clear()
-        for tub in self.tubs:
-            for underlying in tub:
-                record = TubRecord(self.config, tub.base_path, underlying)
-                if not self.train_filter or self.train_filter(record):
-                    self.records.append(record)
-
-        return train_test_split(self.records, shuffle=self.shuffle,
-                                test_size=(1. - self.config.TRAIN_TEST_SPLIT))
+    @staticmethod
+    def is_continuous(rec_1: TubRecord, rec_2: TubRecord) -> bool:
```

Comment:
I think we use `@classmethod` over `@staticmethod`. Use a `cls` as the first parameter if switching over.

Reply:
I think, here '@staticmethod' makes more sense, because it doesn't need a class, it's just a helper function that is used on two `TubRecords`. Because it is used in the implementation of `__iter__()`.  Alternatively, I could make it a member function on `TubRecord` taking another `TubRecord`, that would make sense, too.